src/components/Reservation/HallPlan.vue

```diff
@@ -40,25 +40,35 @@ import { ref } from "vue";
 import CustomButton from "@components/common/CustomButton.vue";
 export default {
   components: { CustomButton },
-  emits: ["selectSeats"],
   setup() {
     const reservationStore = useReservationStore();
-    const { hallMatrix } = storeToRefs(reservationStore);
-    const selectedSeats = ref([]);
-    function toggleSeat(seatNumber) {
-      const seatIndex = selectedSeats.value.findIndex(
-        (element) => element === seatNumber
-      );
-      if (seatIndex < 0) selectedSeats.value.push(seatNumber);
-      else selectedSeats.value.splice(seatIndex, 1);
+    const { hallMatrix, selectedSeatsRaw } = storeToRefs(reservationStore);
+    const selectedSeats = ref(selectedSeatsRaw.value);
+    function toggleSeat(isTaken, seatNumber) {
+      if (!isTaken) {
+        const seatIndex = selectedSeats.value.findIndex(
+          (element) => element === seatNumber
+        );
+        if (seatIndex < 0) selectedSeats.value.push(seatNumber);
+        else selectedSeats.value.splice(seatIndex, 1);
+      }
```

Comment:
Won't it only work now to add a seat if it's not taken and otherwise do nothing? 🤔

Reply:
Ohhhh okay, I misunderstood then 😅
I thought it's a simple boolean toggle - click and boom isTaken, click and not isTaken 😄
Ignore it then 😄